### PR TITLE
Add support for Adafruit Metro RP2040

### PR DESCRIPTION
**Pull request by ThadHouse**
Adds support for https://www.adafruit.com/product/5786 
Copied mostly from the adafruit-itsy-bitsy-rp2040

### CODE REVIEW

File: boards/adafruit-metro-rp2040/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+[package]
+name = "adafruit-metro-rp2040"
+version = "0.1.0"
+authors = ["Thad House <thadhouse1@gmail.com>", "The rp-rs Developers"]
+edition = "2018"
+homepage = "https://github.com/rp-rs/rp-hal-boards/tree/main/boards/adafruit_metro_rp2040"
```
Comment (ahmedcharles) [inline boards/adafruit-metro-rp2040/Cargo.toml:6]:
```suggestion
homepage = "https://github.com/rp-rs/rp-hal-boards/tree/main/boards/adafruit-metro-rp2040"
```

File: boards/adafruit-metro-rp2040/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# [adafruit-metro-rp2040] - Board Support for the [Adafruit Metro RP2040]
+
+You should include this crate if you are writing code that you want to run on
+an [Adafruit Metro RP2040] - an RP2040 board in the Itsy family from Adafruit.
```
Comment (ahmedcharles) [inline boards/adafruit-metro-rp2040/README.md:4]: Copy/paste typo.

File: boards/adafruit-metro-rp2040/README.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+$ cargo install elf2uf2-rs, then repeating the `cargo run` command above.
+```
+
+### [adafruit_metro_blinky](./examples/adafruit_metro_blinky.rs)
+
+Blink an LED on and off.
+
+### [adafruit_metro_rainbow](./examples/adafruit_metro_rainbow.rs)
+
+Continuously changes the color of the ItsyBitsy's onboard Neopixel.
```
Comment (ahmedcharles) [inline boards/adafruit-metro-rp2040/README.md:62]: Copy/paste typo.

File: boards/adafruit-metro-rp2040/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+description = "Board Support Package for the Adafruit Metro RP2040"
+license = "MIT OR Apache-2.0"
+repository = "https://github.com/rp-rs/rp-hal-boards.git"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+cortex-m = "0.7.2"
+rp2040-boot2 = { version = "0.2.0", optional = true }
+rp2040-hal = { version = "0.8.0" }
```
Comment (ahmedcharles) [inline boards/adafruit-metro-rp2040/Cargo.toml:16]: It seems that `rp2040-hal` is now at `0.9.0`, which is not backwards compatible in how `bsp_pins!` is used.

Comment (ThadHouse) [reply boards/adafruit-metro-rp2040/Cargo.toml:16]: Ok. I'll update this and my other PR.

Comment (ThadHouse) [reply boards/adafruit-metro-rp2040/Cargo.toml]: Updated.

**Review by ahmedcharles — APPROVED** [review]
I've tested this locally on a Metro RP2040 and blinky works but rainbow doesn't (I think it uses some things that were updated in a backwards incompatible way).

**Comment by ThadHouse** [comment]
I fixed rainbow.

**Review by ithinuel — APPROVED** [review]
Overall looks good to me.
I cross-checked the pin names and things seem to checkout 👍 

I start feeling like those examples are pretty redundant as we find them on nearly every rp2040 boards.
It'd be nice to have one showcasing the board's specific feature/differenciator.

E.G. something around the SDCard on this one.

**Comment by ThadHouse** [comment]
Agreed on the examples, but they shouldn't be blockers to getting board libraries merged. IMO having examples should be separate requirements from the actual board libraries being created.

**Pull request merged**